packages/instantsearch.js/src/connectors/trending-items/connectTrendingItems.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  createDocumentationMessageGenerator,
+  checkRendering,
+  noop,
+} from '../../lib/utils';
+
+import type { Connector, TransformItems, Hit, BaseHit } from '../../types';
+import type {
+  PlainSearchParameters,
+  RecommendResultItem,
+} from 'algoliasearch-helper';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'trending-items',
+  connector: true,
+});
+
+export type TrendingItemsRenderState<THit extends BaseHit = BaseHit> = {
+  /**
+   * The matched recommendations from the Algolia API.
+   */
+  recommendations: Array<Hit<THit>>;
+};
+
+export type TrendingItemsConnectorParams<THit extends BaseHit = BaseHit> = (
+  | {
+      /**
+       * The facet attribute to get recommendations for.
+       */
+      facetName: string;
+      /**
+       * The facet value to get recommendations for.
+       */
+      facetValue: string;
+    }
+  | { facetName?: never; facetValue?: never }
+) & {
+  /**
+   * The number of recommendations to retrieve.
+   */
+  maxRecommendations?: number;
+  /**
+   * The threshold for the recommendations confidence score (between 0 and 100).
+   */
+  threshold?: number;
+  /**
+   * List of search parameters to send.
+   */
+  fallbackParameters?: Omit<
+    PlainSearchParameters,
+    'page' | 'hitsPerPage' | 'offset' | 'length'
+  >;
+  /**
+   * List of search parameters to send.
```

Comment:
these jsdoc comments aren't that clear, but it can be improved later? What is the description in the api client / current recommend libs?

Reply:
This is the description we used everywhere so far

https://www.algolia.com/doc/api-reference/api-methods/get-frequently-bought-together/#method-param-queryparameters

Reply:
@Haroenv if that's ok we can handle this in FX-2848, we can definitely make the distinction between fallbackParameters and queryParameters clearer.